### PR TITLE
Update project to consume @typespec 

### PR DESCRIPTION
**Pull request by joheredi**
This is the first part of the rename where we are updating dependencies. A follow up PR will come after this with the rename of the emitter and project folders

### CODE REVIEW

File: packages/cadl-typescript/test/commands/gen-cadl-ranch.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import { cadls } from "./cadl-ranch-list.js";
```
Comment (MaryGao) [inline packages/cadl-typescript/test/commands/gen-cadl-ranch.ts:1]: The filename `gen-cadl-ranch` should be `gen-typespec-ranch`. Is it changed this time or next time?

File: packages/cadl-typescript/README.md [hunk 2]
```diff
@@ -26,15 +26,15 @@ Make sure `npx cadl compile .` runs correctly.
 
 ## Add cadl-typescript
 
-Make sure the version of [cadl-typescript release](https://www.npmjs.com/package/@azure-tools/cadl-typescript) depends on the same version of "@cadl-lang/compiler" as in your Cadl project.
```
Comment (MaryGao) [inline packages/cadl-typescript/README.md:29]: The folder name `cadl-typescript` should be `typespe-typescript`. Is it changed this time or next?

File: packages/cadl-rlc-test/package.json [hunk 3]
```diff
@@ -3,16 +3,17 @@
...
   "dependencies": {
-    "@cadl-lang/compiler": "~0.40.0",
+    "@typespec/compiler": "~0.41.0",
     "@azure-tools/cadl-typescript": "1.0.0-beta.10",
-    "@cadl-lang/rest": "~0.40.0",
-    "@cadl-lang/openapi": "~0.40.0",
-    "@cadl-lang/versioning": "~0.40.0",
-    "@azure-tools/cadl-azure-core": "~0.26.0",
-    "@azure-tools/cadl-autorest": "~0.26.0",
-    "@azure-tools/cadl-dpg": "~0.26.0",
```
Comment (MaryGao) [inline packages/cadl-rlc-test/package.json:13]: Just curious, why we still have denpendency for cadl-dpg?
https://github.com/Azure/autorest.typescript/blob/34637ea97bf56924e555e5e88084940a56fb5948/common/config/rush/pnpm-lock.yaml#L159

File: .vscode/launch.json [hunk 4]
```diff
@@ -85,7 +85,7 @@
       "name": "Generate code for CadlRlcTest",
       "request": "launch",
       "cwd": "${workspaceFolder}/packages/cadl-rlc-test/test/openai",
-      "runtimeArgs": ["cadl", "compile", "./spec/main.cadl"],
+      "runtimeArgs": ["cadl", "compile", "./spec/main.tsp"],
```
Comment (lirenhe) [inline .vscode/launch.json:88]: Do we still need to use cadl compile here?

File: packages/cadl-rlc-test/test/anomalyDetector/spec/main.cadl [hunk 5]
```diff
@@ -1,53 +0,0 @@
-import "@cadl-lang/rest";
```
[the remark this answers is not in the document]
Comment (MaryGao) [reply packages/cadl-rlc-test/test/anomalyDetector/spec/main.cadl:1]: Please notice we have a pr to upgrade this: https://github.com/Azure/azure-rest-api-specs/pull/23036

File: .vscode/launch.json [hunk 6]
```diff
@@ -85,7 +85,7 @@
       "name": "Generate code for CadlRlcTest",
```
Comment (MaryGao) [inline .vscode/launch.json:85]:
```suggestion
      "name": "Generate code for TSPRlcTest",

```

File: .vscode/launch.json (hunk 4 above)
Comment (MaryGao) [inline .vscode/launch.json:88]:
```suggestion
      "runtimeArgs": ["tsp", "compile", "./spec/main.tsp"],

```

**Comment by MaryGao** [comment]
Left small comments! I could approve if no missing stuff.

File: packages/cadl-rlc-test/test/anomalyDetector/spec/main.cadl (hunk 5 above)
[the remark this answers is not in the document]
Comment (joheredi) [reply packages/cadl-rlc-test/test/anomalyDetector/spec/main.cadl:1]: Good catch it was unintended

**Pull request merged**